suspect/processing/frequency_correction.py

```diff
@@ -41,26 +61,78 @@ def spectral_registration(data, target, initial_guess=(0.0, 0.0), frequency_rang
     # case we use the spectral points between those two frequencies, or it can
     # be a numpy.array of the same size as the data in which case we simply use
     # that array as the weightings for the comparison
+
+    # Specify some default spectral weights (i.e. equal weighting to all region of the spectrum)
+    spectral_weights = numpy.ones(data.shape)
     if type(frequency_range) is tuple:
-        spectral_weights = frequency_range[0] < data.frequency_axis() & data.frequency_axis() < frequency_range[1]
-    else:
+        spectral_weights = numpy.logical_and(data.frequency_axis() > frequency_range[0],data.frequency_axis() < frequency_range[1])
+    elif frequency_range is not None:
+        # If frequency_range is a numpy array...
         spectral_weights = frequency_range
+    
+    # Apply the spectral weights to the data
+    d = data.spectrum()*spectral_weights
+    t = target.spectrum()*spectral_weights
+    
+    # Convert back to the time-domain MRSData object
+    data = d.fid()
+    target = t.fid()
+    
+    # Check on time_axis constraints
+    # the supplied time range can be None, in which case the whole signal will 
+    # used, or it can be a tuple specifying the lower and upper bounds of the 
+    # time range to use. 
+    if type(time_range) is tuple:
+        # Run the algorithm on a subset of the time domain samples only, defined by the input range
+        time_idx = numpy.logical_and(data.time_axis()>=time_range[0],data.time_axis()<=time_range[1])
+        data = data[time_idx]
+        target = target[time_idx]
+            
 
     # define a residual function for the optimizer to use
     def residual(input_vector):
-        #transformed_data = transform_fid(data, -input_vector[0], -input_vector[1])
         transformed_data = data.adjust_frequency(-input_vector[0]).adjust_phase(-input_vector[1])
         residual_data = transformed_data - target
-        if frequency_range is not None:
-            spectrum = residual_data.spectrum()
-            weighted_spectrum = residual_data * spectral_weights
-            # remove zero-elements
-            weighted_spectrum = weighted_spectrum[weighted_spectrum != 0]
-            residual_data = numpy.fft.ifft(numpy.fft.ifftshift(weighted_spectrum))
         return_vector = numpy.zeros(len(residual_data) * 2)
         return_vector[:len(residual_data)] = residual_data.real
         return_vector[len(residual_data):] = residual_data.imag
         return return_vector
 
     out = scipy.optimize.leastsq(residual, initial_guess)
-    return out[0][0], out[0][1]
+    freq_and_phase_params = (out[0][0], out[0][1])
+    
+    return freq_and_phase_params
+    
+def select_target_for_spectal_registration(data, frequency_range = None):
```

Comment:
I would make an analogy with the `spectral_registration()` method - we chose not to call it `do_frequency_correction()` because there are other types of frequency correction which people might choose to use. By providing multiple different methods we suit different situations. Personally I think that having a set of methods is much better than the method as parameter option (`do_frequency_correction(method="spectral_registration")`) - it keeps both the code and the documentation simpler and makes it easier to pass different arguments to different functions, for example passing the spectrum in some cases and the FID in others.